src/decimal.rs

```diff
@@ -101,6 +101,42 @@ where
     }
 }
 
+impl<I, const D: u8> num_traits::Num for Decimal<I, D> 
+where
+    I: SignedScaledInteger<D>,
+{
+    type FromStrRadixErr = &'static str;
+
+    fn from_str_radix(str: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
+        Ok(Self(I::from_str_radix(str, radix).map_err(|_| "Cannot parse from str")?))
+    }
+}
```

Comment:
I think this conflicts with the current to/from string impl that parses decimal format i.e. `1.250`

Reply:
The implementations of `<I as FromStr>::parse_str()` and `<I as Num>::from_str_radix` for base 10 numbers (`radix == 10`) will be the same output, as the implementation of `Num::from_str_radix` actually just calls the standard methods of the primitive types, e.g: (https://doc.rust-lang.org/std/primitive.i32.html#method.from_str_radix).
Here is where `num_traits` implements `Num::from_str_radix` for the primitives: https://github.com/rust-num/num-traits/blob/022f250f7c12955166dd2ebe776f74e1707312a2/src/lib.rs#L169
So when implementing this method for `Decimal`, we can just use the implementation of the generic type `I`.

Reply:
I think you're right. It should use `Decimal::from_str` inside the `from_str_radix` method, but it also must ensure that `radix == 10` as different numeric bases wouldn't make much sense for a decimal crate.